hidimstat/estimator/_utils/u_Dnn_learner.py

```diff
@@ -0,0 +1,657 @@
+
+    Returns
+    -------
+    X_train_scaled : {array-like, sparse matrix} of shape (n_train_samples, n_features)
+        The sampling_with_repetitionped training input samples with scaled continuous variables.
+    y_train_scaled : {array-like} of shape (n_train_samples, )
+        The sampling_with_repetitionped training output samples scaled if continous.
+    X_validation_scaled : {array-like, sparse matrix} of shape (n_validation_samples, n_features)
+        The validation input samples with scaled continuous variables.
+    y_validation_scaled : {array-like} of shape (n_validation_samples, )
+        The validation output samples scaled if continous.
+    X_scaled : {array-like, sparse matrix} of shape (n_samples, n_features)
+        The original input samples with scaled continuous variables.
+    y_validation : {array-like} of shape (n_samples, )
+        The original output samples with validation indices.
+    scaler_x : Scikit-learn StandardScaler
+        The standard scaler encoder for the continuous variables of the input.
+    scaler_y : Scikit-learn StandardScaler
+        The standard scaler encoder for the output if continuous.
+    valid_ind : list
+        The list of indices of the validation set.
+    """
+    rng = np.random.RandomState(random_state)
+    scaler_x, scaler_y = StandardScaler(), StandardScaler()
+    n = X.shape[0]
+
+    if sampling_with_repetition:
+        train_ind = rng.choice(n, n, replace=True)
+    else:
+        train_ind = rng.choice(
+            n, size=int(np.floor(split_percentage * n)), replace=False
+        )
+    valid_ind = np.array([ind for ind in range(n) if ind not in train_ind])
+
+    X_train, X_validation = X[train_ind], X[valid_ind]
+    y_train, y_validation = y[train_ind], y[valid_ind]
+
+    # Scaling X and y
+    X_train_scaled = X_train.copy()
+    X_validation_scaled = X_validation.copy()
+    X_scaled = X.copy()
+
+    if len(list_continuous) > 0:
+        X_train_scaled[:, list_continuous] = scaler_x.fit_transform(
+            X_train[:, list_continuous]
+        )
+        X_validation_scaled[:, list_continuous] = scaler_x.transform(
+            X_validation[:, list_continuous]
+        )
+        X_scaled[:, list_continuous] = scaler_x.transform(X[:, list_continuous])
+    if problem_type == "regression":
+        y_train_scaled = scaler_y.fit_transform(y_train)
+        y_validation_scaled = scaler_y.transform(y_validation)
+    else:
+        y_train_scaled = y_train.copy()
+        y_validation_scaled = y_validation.copy()
+
+    return (
+        X_train_scaled,
+        y_train_scaled,
+        X_validation_scaled,
+        y_validation_scaled,
+        X_scaled,
+        y_validation,
+        scaler_x,
+        scaler_y,
+        valid_ind,
+    )
+
+
+def sigmoid(x):
+    """
+    This function applies the sigmoid function element-wise to the input array x
+    """
+    return 1 / (1 + np.exp(-x))
+
+
+def softmax(x):
```

Comment:
```suggestion
def _softmax(x):
```

Reply:
The same for this function, it exist in scipy: https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.softmax.html#softmax

Reply:
This will be addressed by the issue #81.